src/clr-angular/utils/loading/loading-listener.ts

```diff
@@ -4,12 +4,10 @@
  * The full license information can be found in LICENSE in the root directory of this project.
  */
 
+import {ClrLoadingState} from "./loading";
+
 /**
  * This is an abstract class because we need it to still be a valid token for dependency injection after transpiling.
  * This does not mean you should extend it, simply implementing it is fine.
  */
-export abstract class LoadingListener {
-    abstract startLoading(): void;
-
-    abstract doneLoading(): void;
-}
\ No newline at end of file
+export abstract class LoadingListener { abstract updateLoadingState(state: ClrLoadingState): void; }
```

Comment:
I know I'm a PITA, but this "update" method feels like calling it will update the loading state itself, but it doesn't. It's what's called _when_ the loading state has been updated.

I feel like `loadingStateChange(state: ClrLoadingState)` shows better what the method should do: update the view based on a new loading state.

Reply:
No problem. Will change the name.